src/main/java/Client/ConsoleClientLauncher.java

```
@@ -63,33 +83,44 @@ public static void stopClient() throws ExecutionException{
             TODO This exception is thrown/printed during socket.close(), no matter if serverShutdown was called before or not...
              */
         } catch (IOException e) {
-            e.printStackTrace();
+            logger.error("error by handling I/O");
         }
         try {
             clientListening.get();
         } catch (InterruptedException ie) {
-            ie.printStackTrace();
+            Thread.currentThread().interrupt();
+            System.exit(1);
```

`System.exit` Is the nuclear option of a controlled crash/exit of the application. It will shut down the *whole* JVM including all running threads without any further reporting of any errors or exceptions beyond the error code (1 in this case). In 99,99% of cases (including this one) this is not what you want. Instead, you should do something like this: `throw new AssertionError("unexpected interrupt", ie);`